### PR TITLE
NEXT-7986 #71 fixed thumbnails for the cms images. Used the default t…

### PR DESCRIPTION
**Pull request by colinmurphy**
…humbnail sizes from the media cms folder settings

### 1. What does this change do, exactly?

It fixes a bug with the cms image block to show thumbnails based off the thumbnail sizes set for the CMS media folder.

### 2. Describe each step to reproduce the issue or behaviour.

1. Create a Shopping Experience
2. Add a image block "image full-sized"
3. Upload an image (preferably above 1920px by 1920px in size)
4. Set the display setting for that block as cover
5. Save Shopping Experience
6. Associate that Shopping Experience with a category
7. When you visit the category page on mobile, the 400px thumbnail image will be loaded
8. When you visit the category page on tablet landscape, the 800px thumbnail image will be loaded
9. When you visit the category page on desktop (below 1920px in width), the 1920px thumbnail image will be loade

### 3. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: changelog/_unreleased/2020-09-17-fix-thumbnails-for-cms-images.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: Fixed thumbnails for cms image block
+issue: NEXT-7986
+author: Colin Murphy
+author_email: colin@studioforty9.com
+author_github: colinmurphy
+---
+# Core
+*  Changed  the template `src/Storefront/Resources/views/storefront/element/cms-element-image.html.twig` to include sizes based off the default sizes set for the CMS media folder.
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-fix-thumbnails-for-cms-images.md:9]:
```suggestion
* Changed the template `src/Storefront/Resources/views/storefront/element/cms-element-image.html.twig` to include sizes based off the default sizes set for the CMS media folder.
```

File: src/Storefront/Resources/views/storefront/element/cms-element-image.html.twig [hunk 2]
```diff
@@ -24,7 +24,12 @@
                                         {% endif %}
 
                                         {% sw_thumbnails 'cms-image-thumbnails' with {
-                                            media: element.data.media
+                                            media: element.data.media,
+                                            'xs': '400px',
```
Comment (tobiasberge) [inline src/Storefront/Resources/views/storefront/element/cms-element-image.html.twig:28]: Does this actually work? In other areas the `xs`, `sm` and so on are inside `sizes` object like this:

```twig
{# src/Storefront/Resources/views/storefront/component/product/card/box-standard.html.twig #}
{% sw_thumbnails 'product-image-thumbnails' with {
    media: cover,
    sizes: {
        'xs': '501px',
        'sm': '315px',
        'md': '427px',
        'lg': '333px',
        'xl': '284px'
    }
} %}
```

And the standard image sizes are not working? Is there anything legit inside the sizes attribute? Usually default sizes are automatically being set by the `thumbnail.html.twig`:

```
{# src/Storefront/Resources/views/storefront/utilities/thumbnail.html.twig #}

{# set image size for every viewport #}
{% set sizes = {
    'xs': (shopware.theme.breakpoint.sm - 1) ~'px',
    'sm': (shopware.theme.breakpoint.md - 1) ~'px',
    'md': ((shopware.theme.breakpoint.lg - 1) / columns)|round(0, 'ceil') ~'px',
    'lg': ((shopware.theme.breakpoint.xl - 1) / columns)|round(0, 'ceil') ~'px'
} %}
```

**Review by tobiasberge — CHANGES_REQUESTED** [review]
Hi @colinmurphy, thanks for your contribution 👍  Can you take a look at: https://github.com/shopwareBoostDay/platform/pull/158/files#r490232622 ?

File: src/Storefront/Resources/views/storefront/element/cms-element-image.html.twig (hunk 2 above)
Comment (colinmurphy) [reply src/Storefront/Resources/views/storefront/element/cms-element-image.html.twig:28]: I could see various sizes created for this in the srcset but I will double check this.

**Review by PaddyS — APPROVED** [review]

File: src/Storefront/Resources/views/storefront/element/cms-element-image.html.twig (hunk 2 above)
Comment (colinmurphy) [reply src/Storefront/Resources/views/storefront/element/cms-element-image.html.twig:28]: @tobiasberge Ah sorry. It was adding the images to the srcset but not actually to the sizes property - https://paste.ec/paste/LMUPpdzM#+JB-24eGl4jySjsyJLcBUz1aR4DXiptgbCA6pX6huxo

I will fix this now.

Comment (tinect) [reply src/Storefront/Resources/views/storefront/element/cms-element-image.html.twig:28]: This hasn't been fixed? :-)

**Review by tinect — CHANGES_REQUESTED** [review]

**Comment by shyim** [comment]
Hey,

the time is left for the Boostday PRs. Feel free to create an updated one at `shopware/platform`. Thanks :pray:

**Pull request closed without merging**